### PR TITLE
HvH armor and vendors

### PR DESCRIPTION
**Pull request by XSlayer300**
## About The Pull Request
Does the following:

- This PR replaces all mentions of M3 armor with PAS. M3 is a CM leftover.
- This PR adds armor variants for all **hostile** ERT for HvH rounds and their job types with it. The values are based on medium armor, medium armor with Tyr 2 and heavy armor with Tyr 2 as they lack req and all of the good stuff they have.
- This PR adds HvH variants for all marine vendors, containing less scoped weapons, shotguns and MGs
- This PR buffs some armors available to PAS level.
 
HvH armor variants for the friendly ERT, deathsquad and CLF (as it is a CM leftover) are not included due to how rare friendly ERTs become hostile ERTs and others because of various reasons.
HvH weapon variants are not included, spawn a HvH weapon vendor instead.

## Why It's Good For The Game
Because no one wants to take time to make vendors meant for HvH so we already have vendors to quickly make some, not to mention the inferior armor of all hostile ERTs as they have no req.

## Changelog
:cl:
add: Added HvH variants for hostile ERT armor and marine vendors, admin-spawnable only.
tweak: Renamed most M3 armor to PAS.
tweak: Removed the suppressor in NEXUS closets, you can still get them via attachment vendor.
balance: Buffs some armor to PAS-level.
/:cl:

### CODE REVIEW

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 1]
```diff
@@ -77,6 +77,41 @@
 /obj/machinery/vending/marine/ex_act(severity)
 	return
 
+// HvH version of the vending machine, containing no snipers or slugs. and MGs, shotguns, grenades and scoped weapons are rarer
+// want to get shotguns, scoped weapons and MGs? go to the cargo vendor instead.
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/marine_vending.dm:81]: You can use the autdoc for these

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 2]
```diff
@@ -115,6 +150,25 @@
 		/obj/item/ammobox/standard_lmg = 2,
 	)
 
+// HvH version of the vending machine, containing no ammo for spec weapons and restricted ones
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/marine_vending.dm:153]: ditto
```suggestion
/// HvH version of the vending machine, containing no ammo for spec weapons and restricted ones
```

File: code/game/objects/machinery/vending/marine_vending.dm [hunk 3]
```diff
@@ -173,7 +227,36 @@
 	. = ..()
 	GLOB.cargo_guns_vendors.Remove(src)
 
-
+//HvH version
+//The only way to get LMG, DMR, shotguns and GPMG
```
Comment (TiviPlus) [inline code/game/objects/machinery/vending/marine_vending.dm:231]: Ditto

File: code/modules/clothing/suits/marine_armor.dm [hunk 4]
```diff
@@ -573,26 +571,43 @@
 	flags_armor_protection = CHEST|GROIN|ARMS|LEGS|FEET|HANDS
 	soft_armor = list("melee" = 55, "bullet" = 60, "laser" = 60, "energy" = 60, "bomb" = 55, "bio" = 10, "rad" = 10, "fire" = 60, "acid" = 60)
 
+// Modified version of the armor for HvH combat. Stats are based on medium armor, with tyr mark 2.
+/obj/item/clothing/suit/storage/faction/UPP/hvh
```
Comment (TiviPlus) [inline code/modules/clothing/suits/marine_armor.dm:575]:
```suggestion
/// Modified version of the armor for HvH combat. Stats are based on medium armor, with tyr mark 2.
/obj/item/clothing/suit/storage/faction/UPP/hvh
```

File: code/modules/clothing/suits/marine_armor.dm [hunk 5]
```diff
@@ -573,26 +571,43 @@
...
+
+
 /obj/item/clothing/suit/storage/faction/UPP/commando
 	name = "\improper UM5CU personal armor"
 	desc = "A modification of the UM5, designed for stealth operations."
 	icon_state = "upp_armor_commando"
 	slowdown = SLOWDOWN_ARMOR_LIGHT
 
+// Modified version of the armor for HvH combat. Stats are based on medium armor, with tyr mark 2.
+/obj/item/clothing/suit/storage/faction/UPP/commando/hvh
```
Comment (TiviPlus) [inline code/modules/clothing/suits/marine_armor.dm:586]:
```suggestion
/// Modified version of the armor for HvH combat. Stats are based on medium armor, with tyr mark 2.
/obj/item/clothing/suit/storage/faction/UPP/commando/hvh
```

File: code/modules/clothing/suits/marine_armor.dm [hunk 6]
```diff
@@ -573,26 +571,43 @@
...
 /obj/item/clothing/suit/storage/faction/UPP/heavy
 	name = "\improper UH7 heavy plated armor"
-	desc = "An extremely heavy duty set of body armor in service with the USL pirates, the UH7 (United Heavy MK5) is known for being a rugged set of armor, capable of taking immesnse punishment."
+	desc = "An extremely heavy duty set of body armor in service with the USL pirates, the UH7 (United Heavy MK7) is known for being a rugged set of armor, capable of taking immesnse punishment."
 	icon_state = "upp_armor_heavy"
 	slowdown = SLOWDOWN_ARMOR_HEAVY
 	soft_armor = list("melee" = 65, "bullet" = 65, "laser" = 65, "energy" = 60, "bomb" = 60, "bio" = 10, "rad" = 10, "fire" = 60, "acid" = 60)
 
+// Modified version of the armor for HvH combat. Stats are based on heavy armor, with tyr mark 2.
+/obj/item/clothing/suit/storage/faction/UPP/heavy/hvh
```
Comment (TiviPlus) [inline code/modules/clothing/suits/marine_armor.dm:597]:
```suggestion
/// Modified version of the armor for HvH combat. Stats are based on heavy armor, with tyr mark 2.
/obj/item/clothing/suit/storage/faction/UPP/heavy/hvh
```

File: code/modules/clothing/suits/marine_armor.dm [hunk 7]
```diff
@@ -573,26 +571,43 @@
...
 /obj/item/clothing/suit/storage/marine/smartgunner/UPP
 	name = "\improper UH7 heavy plated armor"
-	desc = "An extremely heavy duty set of body armor in service with the USL pirates, the UH7 (United Heavy MK5) is known for being a rugged set of armor, capable of taking immesnse punishment."
+	desc = "An extremely heavy duty set of body armor in service with the USL pirates, the UH7 (United Heavy MK7) is known for being a rugged set of armor, capable of taking immesnse punishment."
 	icon_state = "upp_armor_heavy"
 	slowdown = SLOWDOWN_ARMOR_HEAVY
 	soft_armor = list("melee" = 65, "bullet" = 65, "laser" = 65, "energy" = 60, "bomb" = 60, "bio" = 10, "rad" = 10, "fire" = 60, "acid" = 60)
 
+// Modified version of the armor for HvH combat. Stats are based on heavy armor, with tyr mark 2.
+/obj/item/clothing/suit/storage/marine/smartgunner/UPP/hvh
```
Comment (TiviPlus) [inline code/modules/clothing/suits/marine_armor.dm:608]:
```suggestion
/// Modified version of the armor for HvH combat. Stats are based on heavy armor, with tyr mark 2.
/obj/item/clothing/suit/storage/marine/smartgunner/UPP/hvh
```

File: code/modules/clothing/suits/marine_armor.dm [hunk 8]
```diff
@@ -737,14 +750,23 @@
 	flags_item_map_variant = NONE
 
 
+// Modified version of the armor for HvH combat. Stats are on medium armor.
+/obj/item/clothing/suit/storage/marine/som/hvh
```
Comment (TiviPlus) [inline code/modules/clothing/suits/marine_armor.dm:754]:
```suggestion
/// Modified version of the armor for HvH combat. Stats are on medium armor.
/obj/item/clothing/suit/storage/marine/som/hvh
```

File: code/modules/clothing/suits/marine_armor.dm [hunk 9]
```diff
@@ -754,4 +776,8 @@
 	item_state = "som_armor_leader"
 	slowdown = SLOWDOWN_ARMOR_MEDIUM
 	flags_armor_protection = CHEST|GROIN|ARMS|LEGS|HANDS|FEET
-	soft_armor = list("melee" = 55, "bullet" = 50, "laser" = 40, "energy" = 50, "bomb" = 40, "bio" = 15, "rad" = 15, "fire" = 50, "acid" = 60)
+	soft_armor = list("melee" = 55, "bullet" = 50, "laser" = 40, "energy" = 55, "bomb" = 55, "bio" = 55, "rad" = 55, "fire" = 55, "acid" = 60)
+
+// Modified version of the armor for HvH combat. Stats are based on medium armor with mark 2 tyr.
+/obj/item/clothing/suit/storage/marine/som/leader/hvh
```
Comment (TiviPlus) [inline code/modules/clothing/suits/marine_armor.dm:782]:
```suggestion
/// Modified version of the armor for HvH combat. Stats are based on medium armor with mark 2 tyr.
/obj/item/clothing/suit/storage/marine/som/leader/hvh
```

**Review by TiviPlus — APPROVED** [review]

File: code/game/objects/machinery/vending/marine_vending.dm (hunk 1 above)
Comment (XSlayer300) [reply code/game/objects/machinery/vending/marine_vending.dm]: done.

File: code/game/objects/machinery/vending/marine_vending.dm (hunk 3 above)
Comment (XSlayer300) [reply code/game/objects/machinery/vending/marine_vending.dm]: and ditto.

File: code/game/objects/machinery/floodlight.dm [hunk 10]
```diff
@@ -156,7 +156,20 @@
...
+/obj/machinery/floodlight/landing/hvh
+	name = "Sunlight"
+	desc = "Discord light theme edition."
+	density = 0
+	alpha = 0
+	resistance_flags = RESIST_ALL
+
+/obj/machinery/floodlight/landing/hvh/Initialize(mapload, ...)
+	. = ..()
+	set_light(100)
```
Comment (TiviPlus) [inline code/game/objects/machinery/floodlight.dm:175]: Remove this

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/floodlight.dm (hunk 10 above)
Comment (XSlayer300) [reply code/game/objects/machinery/floodlight.dm]: done

**Review by TiviPlus — APPROVED** [review]

**Comment by tgstation-server** [comment]
Your Fix/Feature pull request delta is currently below zero (-8). Maintainers may close future Feature/Tweak/Balance PRs. Fixing issues or helping to improve the codebase will raise this score.

**Pull request merged**